### ISSUE TITLE
plaster 1.1 failed to load development.ini: plaster.exceptions.MultipleLoadersFound

### ISSUE THREAD

**Issue by bostrick** [Pylons/plaster#25]
## Bug Report

**Describe the bug**
Starting with version 1.1, plaster get_loader() raises MultipleLoadersFound on stock installs.

**To Reproduce**
(Stock pyramid install from https://docs.pylonsproject.org/projects/pyramid/en/2.0-branch/narr/project.html#creating-the-project)

```
 $ cookiecutter gh:Pylons/pyramid-cookiecutter-starter --checkout 2.0-branch
 project_name [Pyramid Scaffold]:                                                                                                                                                                                                            
 repo_name [pyramid_scaffold]: breakplaster
 ... choose [1] template_langauge jinja2  and [1] no backend...

$ cd breakplaster/                                                                                                                                                                                                           
$ python3 -m venv env                                                                                                                                                                                             
$ env/bin/pip install --upgrade pip setuptools
...
Successfully installed pip-22.3.1 setuptools-65.5.1

 $ env/bin/pip install -e ".[testing]"
...
Successfully installed Mako-1.2.4 PasteDeploy-3.0.1 Pygments-2.13.0 WebTest-3.0.0 attrs-22.1.0 beautifulsoup4-4.11.1 breakplaster-0.0 coverage-6.5.0 exceptiongroup-1.0.4 hupper-1.10.3 iniconfig-1.1.1 jinja2-3.1.2 markupsafe-2.1.1 packaging-21.3 plaster-1.1 plaster_pastedeploy-1.0.1 pluggy-1.0.0 pyparsing-3.0.9 pyramid-2.0 pyramid-mako-1.1.0 pyramid_debugtoolbar-4.9 pyramid_jinja2-2.10 pytest-7.2.0 pytest-cov-4.0.0 repoze.lru-0.7 soupsieve-2.3.2.post1 tomli-2.0.1 translationstring-1.4 venusian-3.0.0 waitress-2.1.2 webob-1.8.7 zope.deprecation-4.4.0 zope.interface-5.5.1

$ env/bin/pserve development.ini
Traceback (most recent call last):
  File "/home/bowe/breakplaster/env/bin/pserve", line 8, in <module>
    sys.exit(main())
  File "/home/bowe/breakplaster/env/lib64/python3.10/site-packages/pyramid/scripts/pserve.py", line 30, in main
    return command.run()
  File "/home/bowe/breakplaster/env/lib64/python3.10/site-packages/pyramid/scripts/pserve.py", line 189, in run
    loader = self._get_config_loader(config_uri)
  File "/home/bowe/breakplaster/env/lib64/python3.10/site-packages/pyramid/scripts/common.py", line 23, in get_config_loader
    return plaster.get_loader(config_uri, protocols=['wsgi'])
  File "/home/bowe/breakplaster/env/lib64/python3.10/site-packages/plaster/loaders.py", line 115, in get_loader
    raise MultipleLoadersFound(
plaster.exceptions.MultipleLoadersFound: Multiple plaster loaders were found for scheme "file+ini", protocol "wsgi". Please specify a more specific config_uri. Matched loaders: plaster-pastedeploy+file+ini, plaster-pastedeploy+file+ini                                                                                                                                                                                                                                                                                                                                                         
```

**Expected behavior**
plaster properly loads development.ini and starts the web application.

**Additional context**
Not unique to pyramid, plaster 1.1 breaks similarly in my own project as well.  Downgrading to plaster==1.0 resolves the problem.

**Comment by mmerickel** [comment]
Thanks for the report!

I just tested your exact commands on my macOS 13.0.1 machine with python 3.10.8 and the same pip/setuptools as you and didn't run into this. Can you show a `env/bin/pip freeze` maybe and tell me any more about your env so I can repro?

**Comment by mmerickel** [comment]
I tested this in an ubuntu 22.04 docker container as well and didn't run into the issue.

**Comment by bostrick** [comment]
Thanks for rapid response!  system is Fedora36 with python 3.10.8

## python
```
[breakplaster]$ env/bin/python 
Python 3.10.8 (main, Oct 12 2022, 00:00:00) [GCC 12.2.1 20220819 (Red Hat 12.2.1-2)] on linux
...
```

## system information
```
[breakplaster]$ cat /etc/os-release 
NAME="Fedora Linux"
VERSION="36 (Workstation Edition)"
ID=fedora
VERSION_ID=36
VERSION_CODENAME=""
PLATFORM_ID="platform:f36"
PRETTY_NAME="Fedora Linux 36 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:36"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f36/system-administrators-guide/"
SUPPORT_URL="https://ask.fedoraproject.org/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=36
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=36
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
```

## pip freeze output
```
[breakplaster]$ env/bin/pip freeze
attrs==22.1.0
beautifulsoup4==4.11.1
# Editable install with no version control (breakplaster==0.0)
-e /home/bowe/test/breakplaster
coverage==6.5.0
exceptiongroup==1.0.4
hupper==1.10.3
iniconfig==1.1.1
Jinja2==3.1.2
Mako==1.2.4
MarkupSafe==2.1.1
packaging==21.3
PasteDeploy==3.0.1
plaster==1.1
plaster-pastedeploy==1.0.1
pluggy==1.0.0
Pygments==2.13.0
pyparsing==3.0.9
pyramid==2.0
pyramid-debugtoolbar==4.9
pyramid-jinja2==2.10
pyramid-mako==1.1.0
pytest==7.2.0
pytest-cov==4.0.0
repoze.lru==0.7
soupsieve==2.3.2.post1
tomli==2.0.1
translationstring==1.4
venusian==3.0.0
waitress==2.1.2
WebOb==1.8.7
WebTest==3.0.0
zope.deprecation==4.4.0
zope.interface==5.5.1
```

**Comment by mmerickel** [comment]
Can you please show the result of the following in your virtualenv:

```python
import importlib.metadata
print(importlib.metadata.distributions())
```

**Comment by LeResKP** [comment]
Hello @mmerickel,

I have the same issue on my fedora
>>> cat /etc/redhat-release 
Fedora release 36 (Thirty Six)

>>> python --version
Python 3.10.8


>>> pip freeze 
alembic==1.8.1
greenlet==2.0.1
hupper==1.10.3
Jinja2==3.1.2
-e git+ssh://git@github.com/LeResKP/lcg_inventory.git@3205cc25690ea199e5c832f60eccf3480ad2de0b#egg=lcg_inventory
Mako==1.2.4
MarkupSafe==2.1.1
PasteDeploy==3.0.1
Pillow==9.3.0
plaster==1.1
plaster-pastedeploy==1.0.1
psycopg2==2.9.5
Pygments==2.13.0
pyramid==2.0
pyramid-debugtoolbar==4.9
pyramid-jinja2==2.10
pyramid-mako==1.1.0
pyramid-retry==2.1.1
pyramid-tm==2.5
python-barcode==0.14.0
repoze.lru==0.7
SQLAlchemy==2.0.0b3
transaction==3.0.1
translationstring==1.4
typing_extensions==4.4.0
venusian==3.0.0
waitress==2.1.2
WebOb==1.8.7
yapf==0.32.0
zope.deprecation==4.4.0
zope.interface==5.5.2
zope.sqlalchemy==1.6

I changed a bit your command to have better info than itertools.chain object

>>> import importlib.metadata
>>> [vars(d) for d in importlib.metadata.distributions()]
```[{'_path': PosixPath('lcg_inventory.egg-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/setuptools-65.5.1.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/pip-22.3.1.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/python_barcode-0.14.0.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/Pillow-9.3.0.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/yapf-0.32.0.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/psycopg2-2.9.5-py3.10.egg-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/plaster-1.1.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/lcg_inventory.egg-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/zope.sqlalchemy-1.6-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/transaction-3.0.1-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/SQLAlchemy-2.0.0b3-py3.10-linux-x86_64.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/pyramid_tm-2.5-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/pyramid_retry-2.1.1-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/alembic-1.8.1-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/waitress-2.1.2-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/pyramid_debugtoolbar-4.9-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/pyramid_jinja2-2.10-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/pyramid-2.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/plaster_pastedeploy-1.0.1-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/zope.interface-5.5.2-py3.10-linux-x86_64.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/greenlet-2.0.1-py3.10-linux-x86_64.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/typing_extensions-4.4.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/Mako-1.2.4-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/Pygments-2.13.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/repoze.lru-0.7-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/pyramid_mako-1.1.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/zope.deprecation-4.4.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/MarkupSafe-2.1.1-py3.10-linux-x86_64.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/Jinja2-3.1.2-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/WebOb-1.8.7-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/venusian-3.0.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/translationstring-1.4-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/hupper-1.10.3-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib64/python3.10/site-packages/PasteDeploy-3.0.1-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/setuptools-65.5.1.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/pip-22.3.1.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/python_barcode-0.14.0.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/Pillow-9.3.0.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/yapf-0.32.0.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/psycopg2-2.9.5-py3.10.egg-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/plaster-1.1.dist-info')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/zope.sqlalchemy-1.6-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/transaction-3.0.1-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/SQLAlchemy-2.0.0b3-py3.10-linux-x86_64.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/pyramid_tm-2.5-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/pyramid_retry-2.1.1-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/alembic-1.8.1-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/waitress-2.1.2-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/pyramid_debugtoolbar-4.9-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/pyramid_jinja2-2.10-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/pyramid-2.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/plaster_pastedeploy-1.0.1-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/zope.interface-5.5.2-py3.10-linux-x86_64.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/greenlet-2.0.1-py3.10-linux-x86_64.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/typing_extensions-4.4.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/Mako-1.2.4-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/Pygments-2.13.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/repoze.lru-0.7-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/pyramid_mako-1.1.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/zope.deprecation-4.4.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/MarkupSafe-2.1.1-py3.10-linux-x86_64.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/Jinja2-3.1.2-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/WebOb-1.8.7-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/venusian-3.0.0-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/translationstring-1.4-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/hupper-1.10.3-py3.10.egg/EGG-INFO')}, {'_path': PosixPath('/home/lereskp/dev/lcg_inventory/env/lib/python3.10/site-packages/PasteDeploy-3.0.1-py3.10.egg/EGG-INFO')}]```

**Comment by mmerickel** [comment]
Ok so libraries are being installed into both `lib` and `lib64` and for some crazy reason `importlib.metadata.distributions` is returning both copies. That explains the behavior but seems like an importlib bug to me. Regardless it looks like `importlib.metadata.distributions` isn't the best api to be using and switching it up should solve the problem. I'm working on a fix for that.

**Comment by mmerickel** [comment]
Could anyone experiencing this issue please try https://github.com/Pylons/plaster/pull/27?

I was able to repro the problem via explicitly manipulating the PYTHONPATH to add both lib and lib64:

```
$ docker run --rm -it python:3.10 /bin/bash
$ cd /root
$ python3 -m venv env
$ env/bin/pip install plaster-pastedeploy
$ PYTHONPATH=/root/env/lib64/python3.10/site-packages/ env/bin/python
>>> from importlib.metadata import *
>>> from pprint import pprint
>>> pprint([vars(d) for d in distributions()])
[{'_path': PosixPath('/root/env/lib64/python3.10/site-packages/PasteDeploy-3.0.1.dist-info')},
 {'_path': PosixPath('/root/env/lib64/python3.10/site-packages/setuptools-63.2.0.dist-info')},
 {'_path': PosixPath('/root/env/lib64/python3.10/site-packages/plaster-1.1.dist-info')},
 {'_path': PosixPath('/root/env/lib64/python3.10/site-packages/plaster_pastedeploy-1.0.1.dist-info')},
 {'_path': PosixPath('/root/env/lib64/python3.10/site-packages/pip-22.2.2.dist-info')},
 {'_path': PosixPath('/root/env/lib/python3.10/site-packages/PasteDeploy-3.0.1.dist-info')},
 {'_path': PosixPath('/root/env/lib/python3.10/site-packages/setuptools-63.2.0.dist-info')},
 {'_path': PosixPath('/root/env/lib/python3.10/site-packages/plaster-1.1.dist-info')},
 {'_path': PosixPath('/root/env/lib/python3.10/site-packages/plaster_pastedeploy-1.0.1.dist-info')},
 {'_path': PosixPath('/root/env/lib/python3.10/site-packages/pip-22.2.2.dist-info')}]
```

**Comment by LeResKP** [comment]
@mmerickel thanks for the fix, it's working fine with plaster==1.1.2
